src/mscorlib/src/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -82,7 +82,7 @@ private unsafe int GetHashCodeOfStringCore(string source, CompareOptions options
 
             fixed (char* pSource = source)
```

Comment:
CoreCLR doesn't do randomization on the hash when requesting the linguistic hash code. look at the original 

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Globalization/CompareInfo.cs#L1188

it is passing false for the randomization. 
randomized hash done on the string but not on CompareInfo.